python/modules/IcePy/Types.cpp

```diff
@@ -4974,7 +4748,8 @@ IcePy_defineValue(PyObject*, PyObject* args)
     ValueInfoPtr info = lookupValueInfo(id);
     if(!info || info->defined)
     {
-        info = new ValueInfo(id);
+        info = make_shared<ValueInfo>(id);
+        info->init();
```

Comment:
Could/should we change init into a factory method? It would be a bit safer in the off chance we forget to call init.

Reply:
I will create a separate issue as this affects other parts of the C++ code, Slice library, PHP and Ruby mappings and probably more.